src/sentry/models/user.py

```diff
@@ -197,10 +198,13 @@ def class_name(self):
     def delete(self):
         if self.username == "sentry":
             raise Exception('You cannot delete the "sentry" user as it is required by Sentry.')
-        avatar = self.avatar.first()
-        if avatar:
-            avatar.delete()
-        return super().delete()
+        with transaction.atomic(), test_psql_role_override("postgres"):
+            avatar = self.avatar.first()
+            if avatar:
+                avatar.delete()
+            for outbox in User.outboxes_for_update(self.id):
+                outbox.save()
```

Comment:
Should user deletion be a separate outbox message category? It would let us have fewer conditionals in the receiver.

Reply:
Not sure what you mean?  It's the same in all cases.  I also don't think it's wise to separate updates from deletes.  what does the update receiver do if it selects for the object and it no longer exists?

Reply:
>  I also don't think it's wise to separate updates from deletes. what does the update receiver do if it selects for the object and it no longer exists?

Good point, I hadn't considered that scenario.